### PR TITLE
Make the dynamic property writable again

### PR DESCRIPTION
**Pull request by mjauvin**

### CODE REVIEW

**Comment by mjauvin** [comment]
When you get a chance, can you look at the phpstan error @bennothommo ?

File: src/Extension/ExtendableTrait.php [hunk 1]
```diff
@@ -412,12 +403,11 @@ public function extendableSet($name, $value)
...
             $this->extensionCallMethod($parent, '__set', [$name, $value]);
+            return;
         }
 
-        /*
-         * Setting an undefined property
-         */
-        if (!self::$extendableGuardProperties) {
+        // local dynamic property
+        if (array_key_exists($name, $this->getDynamicProperties())) {
```
Comment (bennothommo) [inline src/Extension/ExtendableTrait.php:411]: @mjauvin what's this condition needed for?

**Comment by bennothommo** [comment]
@mjauvin I re-ran the test and it passed. No idea why it failed before :man_shrugging:

File: src/Extension/ExtendableTrait.php (hunk 1 above)
Comment (mjauvin) [reply src/Extension/ExtendableTrait.php:411]: It prevents creating a dynamic property just by setting it (if there's no parent `__set` method for example)

**Comment by mjauvin** [comment]
Ok, added a new unit test to be safe and this now works as expected.

Winter.Translate has been fixed to work properly with new dynamic properties as well.

File: src/Extension/ExtendableTrait.php [hunk 2]
```diff
@@ -412,12 +403,11 @@ public function extendableSet($name, $value)
...
         if ($parent !== false && $this->extensionMethodExists($parent, '__set')) {
             $this->extensionCallMethod($parent, '__set', [$name, $value]);
+            return;
         }
 
-        /*
-         * Setting an undefined property
-         */
-        if (!self::$extendableGuardProperties) {
+        // local dynamic property
```
Comment (LukeTowers) [inline src/Extension/ExtendableTrait.php]:
```suggestion
        // Don't allow automatic creation of dynamic properties through the setter magic method,
        // addDynamicProperty() must be used instead.
```

**Pull request merged**